src/import/FSHTank.ts

```diff
@@ -295,50 +312,64 @@ export class FSHTank implements Fishable {
+              (vs.name === base ||
+                vs.id === base ||
+                getUrlFromFshDefinition(vs, this.config.canonical) === base) &&
+              (version == null || version === getVersionFromFshDefinition(vs, this.config.version))
           );
           if (!result) {
             result = this.getAllInstances().find(
               vsInstance =>
                 vsInstance?.instanceOf === 'ValueSet' &&
                 vsInstance?.usage === 'Definition' &&
-                (vsInstance?.name === item ||
-                  vsInstance.id === item ||
-                  getUrlFromFshDefinition(vsInstance, this.config.canonical) === item)
+                (vsInstance?.name === base ||
+                  vsInstance.id === base ||
+                  getUrlFromFshDefinition(vsInstance, this.config.canonical) === base) &&
+                (version == null ||
+                  version === getVersionFromFshDefinition(vsInstance, this.config.version))
             );
           }
           break;
         case Type.CodeSystem:
           result = this.getAllCodeSystems().find(
             cs =>
-              cs.name === item ||
-              cs.id === item ||
-              getUrlFromFshDefinition(cs, this.config.canonical) === item
+              (cs.name === base ||
+                cs.id === base ||
+                getUrlFromFshDefinition(cs, this.config.canonical) === base) &&
+              (version == null || version === getVersionFromFshDefinition(cs, this.config.version))
           );
           if (!result) {
             result = this.getAllInstances().find(
               csInstance =>
                 csInstance?.instanceOf === 'CodeSystem' &&
                 csInstance?.usage === 'Definition' &&
-                (csInstance?.name === item ||
-                  csInstance.id === item ||
-                  getUrlFromFshDefinition(csInstance, this.config.canonical) === item)
+                (csInstance?.name === base ||
+                  csInstance.id === base ||
+                  getUrlFromFshDefinition(csInstance, this.config.canonical) === base) &&
+                (version == null ||
+                  version === getVersionFromFshDefinition(csInstance, this.config.version))
             );
           }
           break;
         case Type.Instance:
-          result = this.getAllInstances().find(i => i.name === item || i.id === item);
+          result = this.getAllInstances().find(
+            i =>
+              (i.name === base || i.id === base) &&
+              (version == null || version === getVersionFromFshDefinition(i, this.config.version))
+          );
           break;
         case Type.Invariant:
-          result = this.getAllInvariants().find(i => i.name === item);
+          result = this.getAllInvariants().find(
+            i =>
+              i.name === base &&
+              (version == null || version === getVersionFromFshDefinition(i, this.config.version))
+          );
```

Comment:
Invariants don't carry versions, so I don't think this is needed.

Reply:
Part of [0d0852d](https://github.com/FHIR/sushi/pull/1275/commits/0d0852d19cbf5a24e3e63fdf9a7f359ac288bcce)